restapi/src/main/java/io/stargate/web/docsapi/resources/NamespacesResource.java

```diff
@@ -0,0 +1,286 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.web.docsapi.resources;
+
+import com.codahale.metrics.annotation.Timed;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.stargate.db.datastore.DataStore;
+import io.stargate.web.models.Datacenter;
+import io.stargate.web.models.Error;
+import io.stargate.web.models.Keyspace;
+import io.stargate.web.models.ResponseWrapper;
+import io.stargate.web.resources.Converters;
+import io.stargate.web.resources.Db;
+import io.stargate.web.resources.RequestHandler;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.apache.cassandra.stargate.db.ConsistencyLevel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Api(
+    produces = MediaType.APPLICATION_JSON,
+    consumes = MediaType.APPLICATION_JSON,
+    tags = {"documents"})
+@Path("/v2/schemas/namespaces")
+@Produces(MediaType.APPLICATION_JSON)
+public class NamespacesResource {
+
+  private static final Logger logger = LoggerFactory.getLogger(NamespacesResource.class);
+
+  @Inject private Db db;
+  private static final ObjectMapper mapper = new ObjectMapper();
+
+  @Timed
+  @GET
+  @ApiOperation(
+      value = "Get all namespaces",
+      notes = "Retrieve all available namespaces.",
+      response = ResponseWrapper.class,
+      responseContainer = "List")
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 200, message = "OK", response = ResponseWrapper.class),
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal server error", response = Error.class)
+      })
+  public Response getAllNamespaces(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam("X-Cassandra-Token")
+          String token,
+      @ApiParam(value = "Unwrap results", defaultValue = "false") @QueryParam("raw")
+          final boolean raw) {
+    return RequestHandler.handle(
+        () -> {
+          DataStore localDB = db.getDataStoreForToken(token);
+          List<Keyspace> namespaces =
```

Comment:
Should we follow up later with metadata so that only the special keyspaces are returned? For example, probably don't want to return system since you can't do anything with it through the API.